### PR TITLE
Make device::gl_device crate fully self-contained

### LINKED ISSUE #416: Make Device completely self-contained with associated types
This will be required for supporting multiple-backends. Blocked on https://github.com/rust-lang/rust/issues/17307

### PR DESCRIPTION
**Pull request by bjz**
Closes #416.

Sorry for flip-flopping on the type parameter of `Renderer` - I previously changed it before `GlResources` was a 'thing'.

### CODE REVIEW

File: tests/vertex_format.rs [hunk 1]
```diff
@@ -49,8 +49,8 @@ fn test_vertex_format() {
     use secret_lib::gfx;
     use secret_lib::device;
 
-    let buf_vert = device::make_fake_buffer();
-    let buf_inst = device::make_fake_buffer();
+    let buf_vert = device::gl_device::make_dummy_buffer();
```
Comment (kvark) [inline tests/vertex_format.rs]: I'd like these tests to be device-agnostic. We'll clean it up later.

**Comment by kvark** [comment]
:+1: :tada:

**Comment by bjz** [comment]
\o/ party!

File: tests/vertex_format.rs (hunk 1 above)
Comment (bjz) [inline tests/vertex_format.rs]: Agreed.

**Comment by bvssvni** [comment]
:balloon:

**Comment by csherratt** [comment]
:balloon: :tada:

**Pull request merged**